core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -217,17 +225,14 @@ public static RequestBody empty() {
      * @return The created {@code RequestBody}.
      */
     public static RequestBody fromContentProvider(ContentStreamProvider provider, long contentLength, String mimeType) {
```

Comment:
There are places we delegate to these methods (e.g. fromInputStream). Do we need those features to also buffer in memory, or should we move them away from delegating to these methods?

Reply:
Good callout. Done.